projects/wallet-template/src/background/keyring.ts

```diff
@@ -164,31 +232,70 @@ export const createKeyring = () => {
+      assert(currentPassword, "keyring must be unlocked")
       const keystore = await getKeystore()
       assert(keystore, "keyring must be setup")
-      const keysetIndex = keystore.meta.findIndex(
-        ({ type, derivationPaths }) =>
-          // FIXME: support plain accounts
-          type === "keyset" &&
-          derivationPaths.some(
-            (d) => d.chainId === chainId && d.publicKey === publicKey,
-          ),
-      )
-      if (keysetIndex === -1) throw new Error("unknown account")
-      assert(currentPassword, "keyring must be unlocked")
+
+      const keysetIndex = keystore.meta.findIndex((keyset) => {
+        switch (keyset._type) {
+          case "DerivationPathKeystore":
+            return keyset.derivationPaths.some(
+              (d) => d.chainId === chainId && d.publicKey === publicKey,
+            )
+          case "PrivateKeyKeystore":
+            return (
+              privateKeyToPublicKey(keyset.privateKey, keyset.scheme) ===
+              publicKey
+            )
+          default:
+            throw new Error("invalid keystore type")
+        }
+      })
+
+      if (keysetIndex === -1) {
+        throw new Error("unknown account")
+      }
+
       const secret = decodeSecrets(
         keystoreV4.decrypt(keystore, currentPassword),
       )[keysetIndex]
-      const { derivationPaths, scheme } = keystore.meta[keysetIndex]
-      const derivationPath = derivationPaths.find(
-        (d) => d.publicKey === publicKey,
-      )!
-      const createDeriveFn = createDeriveFnMap[scheme]
-      if (!createDeriveFn) throw new Error("invalid signature scheme")
-      return [
-        createDeriveFn(secret)(derivationPath.path),
-        scheme as "Sr25519" | "Ed25519" | "Ecdsa",
-      ] as const
+
+      const keyset = keystore.meta[keysetIndex]
+      switch (keyset._type) {
+        case "DerivationPathKeystore": {
+          const { derivationPaths, scheme } = keyset
+          const derivationPath = derivationPaths.find(
+            (d) => d.publicKey === publicKey,
+          )!
+          const createDeriveFn = createDeriveFnMap[scheme]
+          if (!createDeriveFn) throw new Error("invalid signature scheme")
+          return [
+            createDeriveFn(secret)(derivationPath.path),
+            scheme as "Sr25519" | "Ed25519" | "Ecdsa",
+          ] as const
+        }
+        case "PrivateKeyKeystore": {
+          const { privateKey } = keyset
+          const publicKey = fromHex(
+            privateKeyToPublicKey(keyset.privateKey, keyset.scheme),
+          )
+          let keypair: KeyPair = {
+            publicKey,
+            sign: (msg) => {
+              switch (keyset.scheme) {
+                case "Sr25519":
+                  return sr25519.sign(privateKey, msg)
+                case "Ed25519":
+                  return ed25519.sign(privateKey, msg)
+                case "Ecdsa":
+                  return ecdsa.sign(privateKey, msg)
+              }
+            },
```

Comment:
This should be simplified so the `publicKey` and the `sign` use the same cryptographic `Curve`.

For example
```ts
import {
  sr25519,
  ed25519,
  ecdsa,
  KeyPair,
  Curve,
} from "@polkadot-labs/hdkd-helpers"

const curveFnMap = {
  Sr25519: sr25519,
  Ed25519: ed25519,
  Ecdsa: ecdsa,
} as Record<string, Curve>

const createKeyPair = (privateKey: string, scheme: string): KeyPair => {
  const curve = curveFnMap[scheme]
  if (!scheme) throw new Error("unsupported signature scheme")

  return {
    publicKey: curve.getPublicKey(privateKey),
    sign(message) {
      return curve.sign(message, privateKey)
    },
  }
}